polkadot/node/subsystem-util/src/runtime/mod.rs

```diff
@@ -176,6 +182,26 @@ impl RuntimeInfo {
 		self.get_session_info_by_index(sender, relay_parent, session_index).await
 	}
 
+	/// Get the list of disabled validators at the relay parent.
+	pub async fn get_disabled_validators<Sender>(
+		&mut self,
+		sender: &mut Sender,
+		relay_parent: Hash,
+	) -> Result<Vec<ValidatorIndex>>
+	where
+		Sender: SubsystemSender<RuntimeApiMessage>,
+	{
+		match self.disabled_validators_cache.get(&relay_parent).cloned() {
+			Some(result) => Ok(result),
+			None => {
+				let disabled_validators =
+					get_disabled_validators_with_fallback(sender, relay_parent).await?;
+				self.disabled_validators_cache.insert(relay_parent, disabled_validators.clone());
+				Ok(disabled_validators)
+			},
+		}
+	}
+
```

Comment:
I'm missing something here. Can you explain we need this caching?
We already have one at the [runtime api subsystem](https://github.com/paritytech/polkadot-sdk/blob/99df39194d5de45f46aa107fe5debb49ef2678db/polkadot/node/core/runtime-api/src/cache.rs#L99) which seems to do the same.

Reply:
I guess it's consistency - everything else is cached.

Reply:
See the comments above: https://github.com/paritytech/polkadot-sdk/pull/2637/files#diff-9fd4e4c6eaa75e859a6dfc8a77cb84c1379a6e59413d0c4e5c221621ca1ae31dR76-R79

Reply:
I saw it but I still don't get the benefit - runtime api subsystem will cache the same data, right? The only difference is the message exchange.

Reply:
Yes, but this is the same argument for session index cache.